File: responser-backend/src/main/java/com/responser/backend/model/Review.java
package com.responser.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

/**
 * Review
 *
 * @author Shcherbachenya Igor
 */
@Getter
@Setter
@RequiredArgsConstructor
@Entity
@Table(name = "reviews")
public class Review extends AbstractEntity {

    @Column(name = "user_id")
    private String userId;

    @Column(name = "resource_id")
    private String resourceId;

    @Column(name = "review_id")
    private String reviewId;

    private Byte rating;

    private String text;

    @ManyToOne
    @JoinColumn(name = "resource_id", nullable = false, insertable = false, updatable = false)
    private WebResource webResource;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false, insertable = false, updatable = false)
    private User user;

    @OneToMany(mappedBy = "review", fetch = FetchType.EAGER)
    private Set<ReviewLike> likes;
}

File: reviewly-chrome-extension/src/app/model/Review.ts
import {User} from "./User";
import moment, {Moment} from "moment";
import {ReviewLike} from "./ReviewLike";
import {WebResource} from "./WebResource";
import {ReviewSource} from "./ReviewSource";

export class Review {

    id: string;

    resourceId: string;

    reviewId: string;

    rating: number;

    text: string;

    user: User;

    source: ReviewSource;

    reviewLikes: ReviewLike[];

    webResource: WebResource;

    creationDate: Moment;

    updateDate: Moment;

    public static deserialize(data: any): Review {
        return Object.assign(new Review(), data, {
            user: User.deserialize(data.user),
            creationDate: moment(data.creationDate),
            updateDate: moment(data.updateDate),
            reviewLikes: data.reviewLikes.map((like: any) => ReviewLike.deserialize(like)),
            webResource: data.webResource ? WebResource.deserialize(data.webResource) : null
        });
    }
}